src/DependencyInjection/KnpMenuExtension.php

```diff
@@ -79,9 +79,19 @@ public function prepend(ContainerBuilder $container): void
     private function GetDefinitionsForEnabledProviders($providers, ContainerBuilder $container): void
     {
         foreach ($providers as $builder => $enabled) {
-            if ($enabled) {
-                $container->getDefinition(\sprintf('knp_menu.menu_provider.%s', $builder))->addTag('knp_menu.provider');
-            }
+            $this->GetDefinitionForEnabledProvider($enabled, $container, $builder);
+        }
+    }
+
+    /**
+     * @param $enabled
+     * @param ContainerBuilder $container
+     * @param $builder
+     */
+    private function GetDefinitionForEnabledProvider($enabled, ContainerBuilder $container, $builder): void
```

Comment:
this naming is totally confusing, as it is not getting a definition at all. To me, extracting this method does not even improve code quality. It only makes it harder to see what actually happens.